shared/libraries/opcuatms/opcuatms_server/src/tms_server_context.cpp

```diff
@@ -0,0 +1,28 @@
+#include <opcuatms_server/tms_server_context.h>
+
+BEGIN_NAMESPACE_OPENDAQ_OPCUA_TMS
+
+TmsServerContext::TmsServerContext(const ContextPtr& context)
+    : context(context)
+{
+    this->context.getOnCoreEvent() += event(this, &TmsServerContext::coreEventCallback);
+}
+
+TmsServerContext::~TmsServerContext()
+{
+    this->context.getOnCoreEvent() -= event(this, &TmsServerContext::coreEventCallback);
+}
+
+void TmsServerContext::registerComponent(const ComponentPtr& component, tms::TmsServerObject& obj)
+{
+    idToObjMap.insert(std::make_pair(component.getGlobalId(), obj.weak_from_this()));
```

Comment:
What happens if server component is deleted? I think we should remove it from the map.

Reply:
True, but I think we have issues in OPC UA in general when anything changes on the server structure-wise. Don't think we can currently react to any addition/removal. Maintaining this list isn't really an issue, but I don't see a gain in it at the moment, considering the OPC UA doesn't get deleted in response anyhow.